### ISSUE TITLE
`bevy_render` feature  should depend on `bevy_mikktspace` feature

### ISSUE THREAD

**Issue by jfouche** [jakobhellermann/bevy-inspector-egui#307]
Trying to migrate my game to bevy 0.18, I face a build error in `bevy_impls.rs:136`: 

```rust
#[cfg(feature = "bevy_render")]
impl InspectorPrimitive for Handle<Mesh> {
...

        if ui.button("Generate tangents").clicked() {
            let _ = mesh.generate_tangents();
        }
```

```error 
no method named `generate_tangents` found for mutable reference `&mut bevy_mesh::Mesh` in the current scope
method not found in `&mut bevy_mesh::Mesh`
```

In bevy_mesh_0.18.0, the `generate_tangents` function is under `bevy_mikktspace` feature

Adding `bevy_mikktspace` in my game `cargo.toml` file fixed the issue, but I suppose it should be done in bevy-inspector-egui crate.

**Comment by taboky-dev** [comment]
Oof, I didn't catch this.

From what I can tell, by default the `bevy_mikktspace` feature in `bevy_mesh` is enabled by `bevy_pbr`, which is present in the Cargo.toml:

```toml
# crates/bevy-inspector-egui/Cargo.toml
[features]
default = [
    "bevy",
    # ...
]
# Enable inspector UI for all optional bevy types
bevy = [
    "bevy_pbr", # <-- This enables `bevy_mesh/bevy_mikktspace` by default
    "bevy_render" # <-- If enabled, the trait that calls the `generate_tangets` function is implemented
    # ...
]
bevy_render = ["dep:bevy_render", "bevy_egui/render"] # The trait is implemented, but doesn't guarantee `bevy_pbr` is present or `bevy_mesh/bevy_mikktspace` is enabled
```

By disabling `default-features` from `bevy-inspector-egui` and enabling the feature `bevy_render`, if bevy doesn't enable `bevy_mesh/bevy_mikktspace`, the build error happens. Bevy now recommends disabling their default features and enabling either `2d` or `3d` (or some other group), their `2d` feature doesn't include `bevy_pbr` or any dependency that enables `bevy_mesh/bevy_mikktspace` (since this is only a `3d` thing).

So the following works:
```toml
[dependencies]
bevy = { version = "0.18.0", default-features = false, features = ["2d"] } # Does not include `bevy_pbr` nor enables `bevy_mesh/bevy_mikktspace`
bevy-inspector-egui = "0.36.0" # It includes `bevy_pbr` which enables `bevy_mesh/bevy_mikktspace`
bevy_egui = "0.39.0"
```

But this doesn't:
```toml
[dependencies]
bevy = { version = "0.18.0", default-features = false, features = ["2d"] }
bevy-inspector-egui = { version = "0.36.0", default-features = false, features = ["bevy_render"] }
bevy_egui = "0.39.0"
```

Quickest fix is adding `"bevy_mesh/bevy_mikktspace"` to the `bevy_render` feature, which I found is the only problematic feature when `default-features` is disabled:
```toml
# crates/bevy-inspector-egui/Cargo.toml

# Show inspector UI for bevy_render types
bevy_render = ["dep:bevy_render", "bevy_egui/render", "bevy_mesh/bevy_mikktspace"]
```

**Comment by taboky-dev** [comment]
I may have pointed out the obvious with my previous comment, I am just not sure if it is fine to add `3d`-related features/dependencies to `2d`-only apps/games or vice-versa.

**Comment by Person-93** [comment]
The mikktspace crate is pary of bevy's `3d_api` collection. It shouldn't be brought in if people are only using 2d rendering.
